redi-connect-front/src/pages/front/signup/SignUp.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from "react";
+import * as Yup from "yup";
+
+import { Formik, FormikValues, FormikActions } from "formik";
+import Grid from "@material-ui/core/Grid";
+import {
+  TextField,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+} from "@material-ui/core";
+import omit from "lodash/omit";
+
+import { Button, FormControlLabel, Checkbox } from "@material-ui/core";
+
+import { signUp } from "../../../services/api/api";
+import { RedProfile } from "../../../types/RedProfile";
+import { history } from "../../../services/history/history";
+
+import { courses } from "../../../config/config";
+
+export const validationSchema = Yup.object({
+  firstName: Yup.string()
+    .required()
+    .max(255),
+  lastName: Yup.string()
+    .required()
+    .max(255),
+  username: Yup.string()
+    .email()
+    .label("Email")
+    .max(255),
+  password: Yup.string()
+    .min(8, "Password must contain at least 8 characters")
+    .required("Enter your password")
+    .label("Password"),
+  passwordConfirm: Yup.string()
+    .required("Confirm your password")
+    .oneOf([Yup.ref("password")], "Password does not match"),
+  agreesWithCodeOfConduct: Yup.boolean()
+    .required()
+    .oneOf([true]),
+  gaveGdprConsentAt: Yup.string()
+    .required()
+    .label("Data usage consent"),
+  mentee_currentlyEnrolledInCourse: Yup.string().when("formType", {
+    is: "public-sign-up-mentee-pending-review",
+    then: Yup.string()
+      .oneOf(courses.map(level => level.id))
+      .label("Currently enrolled in course"),
+  }),
+});
+
+export type SignUpFormType =
+  | "mentor"
```

Comment:
I think we should remove "mentor" and "mentee" here?

Reply:
sure, was not sure if it is need anywhere. removed.

Reply:
That's the beauty of TS! You remove stuff here, see if anything breaks :)